cap-ci/pipeline.template

```diff
@@ -1,21 +1,23 @@
 ---
+{{ $config := ( tmpl.Exec "config" ) | data.JSON}}
 
 groups:
 {{ range $_, $Backend := (ds "BACKEND").backend -}} # for all backends
 - name: {{$Backend}}
   jobs:
-{{ range $_, $Jobs := slice "deploy" "smoke-tests" "cf-acceptance-tests-brain" "cf-acceptance-tests" "sync-integration-tests" "upgrade" "stratos" -}} # for jobs
+
+{{ range $_, $Jobs := slice "create-k8s" "deploy" "smoke-tests" "cf-acceptance-tests-brain" "cf-acceptance-tests" "sync-integration-tests" "upgrade" "stratos" -}} # for jobs
 {{ range $_, $EiriniFlag := (ds "EIRINI").eirini -}} # for eirini
 {{ range $_, $OptionsFlag := (ds "OPTIONS").options -}} # for options
     - {{$Jobs}}-{{$EiriniFlag}}-{{$Backend}}-{{$OptionsFlag}}
 {{ end }} # end jobs
 {{ end }} # end eirini
-{{ end }} # end options
+{{ end }} # end OptionsFlag
 {{ end }} # end backend
 - name: ALL
   jobs:
 {{ range $_, $Backend := (ds "BACKEND").backend -}} # for all backends
-{{ range $_, $Jobs := slice "deploy" "smoke-tests" "cf-acceptance-tests-brain" "cf-acceptance-tests" "sync-integration-tests" "upgrade" "stratos" -}} # for jobs
+{{ range $_, $Jobs := slice "create-k8s" "deploy" "smoke-tests" "cf-acceptance-tests-brain" "cf-acceptance-tests" "sync-integration-tests" "upgrade" "stratos" -}} # for jobs
```

Comment:
> This also fails if `create_k8s` is false.
> 
it won't fail. it will point out "create-k8s" as a job even if its set to "false". I did not change it as I have no idea why we need a list of jobs in the first place.

> I keep thinking that it would be way easier to have a pipeline that just consumes from a `clean` pool, and returns to a `deployed` pool if successful, or a `dirty` pool if failed.
> 
> Then creation and deletion is clearer and not buggy.

we don't need to maintain a state at pools level. state is handled at CI level. if one uses create k8s job then one will get a clean k8s. If not, then CI has ability to grab a specific cluster (which may be new or old- depends on user)

Reply:
> > This also fails if `create_k8s` is false.
> 
> it won't fail. it will point out "create-k8s" as a job even if its set to "false". I did not change it as I have no idea why we need a list of jobs in the first place.
> 

It fails:
```
apply configuration? [yN]: y
error: invalid pipeline config:
invalid groups:
        group 'caasp4' has unknown job 'create-k8s-diego-caasp4-sa'
        group 'caasp4' has unknown job 'create-k8s-diego-caasp4-ha'
        (30 lines snipped..)
```


> we don't need to maintain a state at pools level. state is handled at CI level. if one uses create k8s job then one will get a clean k8s.

You are deciding that state is handled at CI level, yet is not enough for what is needed.

> If not, then CI has ability to grab a specific cluster (which may be new or old- depends on user)

Just add `clean`, `dirty`, `completed` folders to the pool. It's a conclusion that will appear itself once you have pets and cattle clusters from both public cloud and internal resources.